### PR TITLE
Refresh contacts on location change with iPad. Addresses #153.

### PR DESCRIPTION
**Pull request by erikkrietsch**
I couldn't separate the logic like I mentioned doing in issue #153, so I wound up removing the observer when setting the location via `EditLocationViewController` then add it back once that action is completed.

### CODE REVIEW

File: FiveCalls/FiveCalls/IssueDetailViewController.swift [hunk 1]
```diff
@@ -56,6 +59,36 @@ class IssueDetailViewController : UIViewController, IssueShareable {
         tableView.reloadRows(at: tableView.indexPathsForVisibleRows ?? [], with: .none)
     }
 
+    func changedLocation() {
+      didUpdateLocation()
+    }
+
+    func didUpdateLocation(using viewController: UIViewController? = nil) {
+        issuesManager.fetchIssues { result in
+            if let vc = viewController, self.issuesManager.isSplitDistrict {
```
Comment (mccarron) [inline FiveCalls/FiveCalls/IssueDetailViewController.swift]: I think we should `[weak self]` these self references.

File: FiveCalls/FiveCalls/IssueDetailViewController.swift [hunk 2]
```diff
@@ -56,6 +59,36 @@ class IssueDetailViewController : UIViewController, IssueShareable {
...
+            } else {
+                viewController?.dismiss(animated: true, completion: nil)
+            }
+
+            if let issue = self.issuesManager.issue(withId: self.issue.id) {
+                self.issue = issue
+                self.tableView.reloadSections([IssueSections.contacts.rawValue], with: .automatic)
+                self.scrollToBottom()
+            } else {
+                // weird state to be in, but the issue we're looking at
```
Comment (mccarron) [inline FiveCalls/FiveCalls/IssueDetailViewController.swift]: Thanks for the comment helpful for looking at this later.

**Comment by subdigital** [comment]
Thanks for putting this together, and yeah... yikes, this isn't ideal.

My first thought is that the issues container vc is reaching directly into the issues vc to tell it to `loadIssues()`, and this is the crux of the issue.

There are multiple visible parties who need to respond to this event, so I think this should be reworked.

So what if we moved that responsibility up into the IssuesContainerViewController, and once it is done it can send a notification that issues were loaded?

Issues VC then responds by updating it's array / table view, and the detail VC responds by pulling out the relevant issue and rebinding the UI.

There might be some kinks we'll run into implementing this that I'm not seeing, but I think this is probably better than turning on/off notification handlers.

Thoughts welcome.

**Comment by erikkrietsch** [comment]
Yeah, certainly agreed that this solution isn’t ideal. 

I had some similar thoughts, although my initial instinct was to get any location responsibility up out of the `IssueDetailViewController` and probably push that into `EditLocationViewController`. 

To be honest, I wasn’t sure how much leeway I had to operate, and didn’t want to overstep the bounds of this particular issue. But since you’ve broached the subject, I’ll consider a more holistic solution.

You may also be right that the `IssuesContainerViewController` can handle these responsibilities and push issues into the detail controller, though I’m not sure how the iPhone layout would respond when the user is in a split district. I’ll try out a couple things and see what I can come up with.

**Comment by subdigital** [comment]
Completely understand about not wanting to increase the scope of the issue. I think it's worth it to improve some of these awkward interactions. Thanks for taking the time to look into this! 🙏

**Pull request closed without merging**